hw31/js/snake.js

```diff
@@ -0,0 +1,232 @@
+                    row: row - 1,
+                };
+            } break;
+            case D.DOWN: {
+                snakePartToShift = {
+                    cell,
+                    row: row + 1,
+                };
+            } break;
+        }
+
+        this.#snake.unshift(snakePartToShift);
+
+        this.#clear();
+        this.#update();
+    }
+
+    #update() {
+        for( let [index, snakePart] of this.#snake.entries()) {
+            let cellElement = this.#findByCoords(snakePart);
+
+            if(index === 0) {
+                cellElement.classList.add(Snake.snakeHeadCssClass, Snake.snakeCssClass);
+
+                this.#head = cellElement;
+
+                continue;
+            }
+
+            cellElement.classList.add(Snake.snakeBodyCssClass, Snake.snakeCssClass);
+        }
+
+        this.#checkIfSnakeHasEaten();
+        this.#checkOnTailCrash();
+    }
+
+    #updateDirection(event) {
+        let key = event.key;
+
+        if (key === 'ArrowLeft' && this.direction !== D.RIGHT) {
+            this.direction = D.LEFT;
+        } else if(key === 'ArrowUp' && this.direction !== D.DOWN) {
+            this.direction = D.UP;
+        } else if(key === 'ArrowRight' && this.direction !== D.LEFT) {
+            this.direction = D.RIGHT;
+        } else if(key === 'ArrowDown' && this.direction !== D.UP) {
+            this.direction = D.DOWN;
+        }
+    }
+
+    #noWallMode({ cell, row }) {
+        if (row === 0 && this.direction === D.UP) {
+            row = this.gridCount;
+        } else if(cell === 0 && this.direction === D.LEFT) {
+            cell = this.gridCount;
+        } else if(row === this.gridCount - 1 && this.direction === D.DOWN) {
+            row = -1;
+        } else if(cell === this.gridCount - 1 && this.direction === D.RIGHT) {
+            cell = -1;
+        }
+
+        return { cell, row };
+    }
+
+    #buildSnake(startCell, startRow, size = 5) {
+        return new Array(size).fill(null).map((_value, index) => {
+            return { cell: startCell + index, row: startRow };
+        })
+    }
+
+    #generateFood() {
+        const getRandomPoint = () => Math.floor(Math.random() * this.gridCount);
+        const foodCoords = { cell: getRandomPoint(), row: getRandomPoint() }
+
+        this.#food = this.#findByCoords(foodCoords);
+        this.#food.append(this.#foodImage);
+    }
+
+    #checkIfSnakeHasEaten() {
+        if (this.#food === this.#head) {
+            this.#removeFood();
+            this.#generateFood();
+            this.#increaseScore();
+        } else {
+            this.#snake.pop();
+        }
+    }
+
+    #checkOnTailCrash() {
+        if (this.#head.classList.contains(Snake.snakeBodyCssClass)) {
+            this.#endGame();
+        }
+    }
+
+    #clear() {
+        let cells = this.find(`.${Snake.snakeCssClass}`);
+
+        cells.forEach(cell => {
+            cell.className = Snake.snakeCellCssClass;
+        });
+    }
+
+    #endGame() {
```

Comment:
The score value exists in DOM after the game is over. 

<img width="683" alt="image" src="https://user-images.githubusercontent.com/24583373/226762678-78593300-0c78-4036-921b-8729c3294c97.png">